### ISSUE TITLE
Extracting node bounding-box coordinates

### ISSUE THREAD

**Issue by arthurflor23** [Jollywatt/typst-fletcher#156]
Hi,

Is there any way to get the position and size of each node in a diagram? For example, rendering a diagram and also generating a file with the bounding box coordinates of each node.

Is it possible to do this? Or what would be the recommended approach?

**Comment by Jollywatt** [comment]
This is an interesting question. It is possible to do this in v0.5.8 by overriding the `render` callback in `diagram()` like so, and storing the info in metadata elements.
```typ
#import "@preview/fletcher:0.5.8" as fletcher: diagram, node, edge, cetz

#diagram($
  A edge("->") edge("d", "=") & B \
  X Y Z edge("tr", "--")
$, render: (grid, nodes, edges, options) => {
  cetz.canvas(fletcher.draw-diagram(grid, nodes, edges, debug: options.debug))
  metadata(nodes)
})

#context query(metadata).first().value.map(node => {
  (center: node.pos.xyz, size: node.size)
})
```

This won't work (at least not in the same way) in v0.6.0, which I'm currently working on.
May I ask what the larger goal is that you want to achieve here? I may consider adding something to allow this kind of introspection in v0.6.0 if it is really needed.

**Comment by arthurflor23** [comment]
Thank you, this is very helpful!

The main goal is to use Fletcher to automatically generate diverse diagrams as training data for deep learning detection models. For that, we would render the diagrams as PNG images, but we also need ground-truth annotations for each element, including its id/name, position, size etc.

In practice, these values would be used to generate bounding boxes for the diagram elements, which would serve as labels for object detection models.
